### ISSUE TITLE
"etree" is unknown import symbol (lxml)

### ISSUE THREAD

**Issue by cebor** [microsoft/pylance-release#392]
## Environment data

-   Language Server version: 2020.9.5
-   OS and version: macOS 10.15.6 (19G2021)
-   Python version (& distribution if applicable, e.g. Anaconda): python3.8 (venv)

## Actual behaviour

I use the library `lxml:4.5.2` and getting following error while importing, also autocomplete is not working. (https://lxml.de/)

```python
from lxml import etree
```

```
"etree" is unknown import symbol Pylance (reportGeneralTypeIssues)
```

With language server set to `jedi` all is working fine.

**Comment by jakebailey** [comment]
lxml's `etree` module appears to be compiled, so we can't get any information out of it as there's no source to read (and we currently don't have a scraper to attempt to load the module and inspect it at runtime). At the moment, the best solution would be to have a set of stubs for `lxml` (as it appears to be almost entirely compiled code).

Note that `reportGeneralTypeIssues` only appears if your type checking mode is set to something other than `"off"`, and off is the default for Pylance.

**Comment by cebor** [comment]
Installing the correct stubs fixed it for me, thanks for the hint.

https://github.com/lxml/lxml-stubs

```bash
pip install lxml-stubs
```

**Comment by huguesv** [comment]
Pylance is able to make use of stubs packages, and there is one available for lxml: `pip install lxml-stubs`

It might work okay. I don't get any error on this simple code (in basic type checking mode):

![image](https://user-images.githubusercontent.com/1696845/93636084-26437b00-f9a8-11ea-8b92-63f664048e9d.png)

**Comment by jakebailey** [comment]
Thanks for the tip; I didn't realize those had existed in typeshed but had been removed. We might be able to bundle them as we do with a few other commonly used libraries if lxml is common enough.

**Comment by ssbarnea** [comment]
I fact I suspect pylance ability to import submodules is partially broken because I see similar errors when importing submodules from my own package. Neither mypy, pylint or flake8 complain about them but pylance wrongly produces them.

**Comment by jakebailey** [comment]
This issue is about `lxml.etree` not having any members; we have resolved this in the last release via special compiled coded stubs. @ssbarnea If you're having issues resolving imports, then please file a new issue. I'd be interested to see if downgrading Pylance fixes your issue.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.4.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202140-7-april-2021
